### ISSUE TITLE
[Request]: [SLOs] Update SLO privilege model docs

### ISSUE THREAD

**Issue by dominiqueclarke** [elastic/observability-docs#4229]
### What documentation page is affected

https://www.elastic.co/guide/en/observability/current/slo-privileges.html

### What change would you like to see?

The SLO privilege model was updated for 8.16.0.

In particular, there is no longer a requirement for SLO editor users to have cluster privileges `manage_transform` and `manage_ingest_pipelines`. 

Additionally, the correct index privileges were added to the Observability `editor` role. Therefore, users can now choose to either use the `editor` role or create a custom role to grant index privileges to SLO users.

The required index privileges are as follows

Editor user
```
.slo-observability-*: `read`, `view_index_metadata`, `write`, `manage`
```
Viewer user
```
.slo-observability-*: `read`, `view_index_metadata`

### Additional info

_No response_

**Comment by eedugon** [comment]
Proposed changes to the doc (already working on them):

- Change the names `SLO All` and `SLO Read` for the roles to become `SLO Editor` and `SLO Viewer`.
- Update the permissions needed for both types of roles to match the new values.
- Update screenshots.

Add the information about the `editor` **built-in role** being another option to grant SLO Editor permissions to the users.

Any thoughts?

**Comment by eedugon** [comment]
PR draft started here: https://github.com/elastic/observability-docs/pull/4300

**Comment by eedugon** [comment]
@dominiqueclarke , would we need to update anything in the `serverless` documentation related with this issue?

Based on the doc https://www.elastic.co/docs/current/serverless/observability/create-an-slo it's mentioned that `The Admin role or higher is required to create SLOs.`, so I would say we don't need to change anything at Serverless documentation side.

By the way, is the previous statement accurate for serverless? Shouldn't `Editor` be enough based on the explanations from https://www.elastic.co/docs/current/serverless/general/assign-user-roles#observability ? (sorry for the offtopic).

**Comment by dominiqueclarke** [comment]
@eedugon 

> is the previous statement accurate for serverless? Shouldn't Editor be enough based on the explanations from

You are correct. We should update the serverless docs to mention `The Editor role or higher is required to create SLOs`, rather than `Admin`.

**Comment by eedugon** [comment]
thanks @dominiqueclarke : I have updated also the serverless doc in the mentioned PR
